solana/programs/matching-engine/src/fallback/processor/initialise_fast_market_order.rs

```diff
@@ -0,0 +1,199 @@
+            AccountMeta::new_readonly(*self.verify_vaa_shim_program, false),
+            AccountMeta::new_readonly(*self.system_program, false),
+        ]
+    }
+}
+
+#[derive(Debug, Copy, Clone, Pod, Zeroable)]
+#[repr(C)]
+pub struct InitialiseFastMarketOrderData {
+    /// The fast market order as the bytemuck struct
+    pub fast_market_order: FastMarketOrderState,
+    /// The guardian set bump
+    pub guardian_set_bump: u8,
+    /// Padding to ensure bytemuck deserialization works
+    _padding: [u8; 7],
+}
+impl InitialiseFastMarketOrderData {
+    // Adds the padding to the InitialiseFastMarketOrderData
+    pub fn new(fast_market_order: FastMarketOrderState, guardian_set_bump: u8) -> Self {
+        Self {
+            fast_market_order,
+            guardian_set_bump,
+            _padding: [0_u8; 7],
+        }
+    }
+
+    /// Deserializes the InitialiseFastMarketOrderData from a byte slice
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - A byte slice containing the InitialiseFastMarketOrderData
+    ///
+    /// # Returns
+    ///
+    /// Option<&Self> - The deserialized InitialiseFastMarketOrderData or None if the byte slice is not the correct length
+    pub fn from_bytes(data: &[u8]) -> Option<&Self> {
+        bytemuck::try_from_bytes::<Self>(data).ok()
+    }
+}
+
+pub struct InitialiseFastMarketOrder<'ix> {
+    pub program_id: &'ix Pubkey,
+    pub accounts: InitialiseFastMarketOrderAccounts<'ix>,
+    pub data: InitialiseFastMarketOrderData,
+}
+
+impl InitialiseFastMarketOrder<'_> {
+    pub fn instruction(&self) -> Instruction {
+        Instruction {
+            program_id: *self.program_id,
+            accounts: self.accounts.to_account_metas(),
+            data: FallbackMatchingEngineInstruction::InitialiseFastMarketOrder(&self.data).to_vec(),
+        }
+    }
+}
+
+/// Initialises the fast market order account
+///
+/// The verify shim program first checks that the digest of the fast market order is correct, and that the guardian signature is correct and recoverable.
+/// If this is the case, the fast market order account is created. The fast market order account is owned by the matching engine program. It can be closed
+/// by the close fast market order instruction, which returns the lamports to the close account refund recipient.
+///
+/// # Arguments
+///
+/// * `accounts` - The accounts of the fast market order and the guardian set
+///
+/// # Returns
+///
+/// Result<()>
+pub fn initialise_fast_market_order(
+    accounts: &[AccountInfo],
+    data: &InitialiseFastMarketOrderData,
+) -> Result<()> {
+    require_min_account_infos_len(accounts, 6)?;
+
+    let signer = &accounts[0];
+    let fast_market_order_account = &accounts[1];
+    let guardian_set = &accounts[2];
+    let guardian_set_signatures = &accounts[3];
+
+    let InitialiseFastMarketOrderData {
+        fast_market_order,
+        guardian_set_bump,
+        _padding: _,
+    } = *data;
```

Comment:
Can this instead just use the reference to data? It should be more efficient to just reference the `fast_market_order` when serializing the bytes into an account (because the ref yields the byte slice).